runner/runner.go

```diff
@@ -19,14 +19,18 @@ const (
 	// States below are end states
 	// a Process in an end state will not change its state
 
-	// Ran to completion
+	// Run completed, possibly with nonzero exit code.
 	COMPLETE
-	// Could not run to completion
+	// Could not complete, bad request or golang runtime err.
```

Comment:
I think we should differentiate between bad request and failed to run.  Having this distinction will help us abort sending poison pill messages around, or falsely marking nodes as dead

BadRequest means it should not be retried by the caller. 
FailedToRun is a retryable error.